### PR TITLE
docs: small fixes, update link checking instructions

### PR DESCRIPTION
**Pull request by sys3175**
fixed two typos (well, one of them wasn't really a typo, but is undocumented, so I fixed it) and added two links for convenience

### Motivation

  * This PR fixes typos/inconsistencies and adds convenience.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any
  [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: doc/user/CONTRIBUTING.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ to `git pull` the latest version of Materialize.
 
     ```shell
     cd doc/user
-    hugo serve -D
+    hugo server -D
     ```
```
Comment (msudberg) [inline doc/user/CONTRIBUTING.md:44]: "hugo serve" is a supported alias for "hugo server", so they do the same thing.

**Review by msudberg — APPROVED** [review]
lgtm!

File: doc/user/README.md [hunk 2]
```diff
@@ -59,9 +59,10 @@ You can now include the diagram on a page using:
 
 ### Checking links locally
 
-You can check links locally with `htmltest`.
+You can check links locally with [htmltest](https://github.com/wjdp/htmltest).
```
Comment (morsapaes) [inline doc/user/README.md:62]:
```suggestion
You can check links locally with [`htmltest`](https://github.com/wjdp/htmltest).
```

**Comment by morsapaes** [comment]
Retriggered the build to work around failing checks. Thanks, @sys3175. 👍

**Pull request merged**